Crash when trying to export the same member twice

## Compiler version

Latest `main` to date, commit `a5a85a28218d1a869680c33cf2dc09e1bf7be829`.

## Minimized code

```scala
object M {
  type B
}

export M.{B, B}
```

## Output (click arrow to expand)
<details>

```scala
Exception in thread "main" java.lang.AssertionError: assertion failed: duplicate type B#4873; previous was type B#4872                                                                                                                                                                                      
        at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
        at dotty.tools.dotc.core.Scopes$MutableScope.enter(Scopes.scala:271)
        at dotty.tools.dotc.core.SymDenotations$ClassDenotation.enterNoReplace(SymDenotations.scala:1962)
        at dotty.tools.dotc.core.SymDenotations$ClassDenotation.enter(SymDenotations.scala:1953)
        at dotty.tools.dotc.core.Symbols$Symbol.entered(Symbols.scala:206)
        at dotty.tools.dotc.typer.Namer.dotty$tools$dotc$typer$Namer$ClassCompleter$$_$process$1$$anonfun$1(Namer.scala:1267)
        at scala.collection.immutable.List.foreach(List.scala:333)
        at dotty.tools.dotc.typer.Namer$ClassCompleter.process$1(Namer.scala:1267)
        at dotty.tools.dotc.typer.Namer$ClassCompleter.processExports(Namer.scala:1278)
        at dotty.tools.dotc.typer.Namer$ClassCompleter.completeInCreationContext(Namer.scala:1463)
        at dotty.tools.dotc.typer.Namer$Completer.complete(Namer.scala:808)
        at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:167)
        at dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:187)
        at dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:189)
        at dotty.tools.dotc.core.SymDenotations$SymDenotation.ensureCompleted(SymDenotations.scala:369)
        at dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:2602)
        at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2865)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2895)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2960)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2964)
        at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3080)
        at dotty.tools.dotc.typer.TyperPhase.typeCheck$$anonfun$1(TyperPhase.scala:43)
        at dotty.tools.dotc.core.Phases$Phase.monitor(Phases.scala:414)
        at dotty.tools.dotc.typer.TyperPhase.typeCheck(TyperPhase.scala:50)
        at dotty.tools.dotc.typer.TyperPhase.runOn$$anonfun$3(TyperPhase.scala:84)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
        at scala.collection.immutable.List.foreach(List.scala:333)
        at dotty.tools.dotc.typer.TyperPhase.runOn(TyperPhase.scala:84)
        at dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:225)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
        at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1328)
        at dotty.tools.dotc.Run.runPhases$1(Run.scala:236)
        at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:244)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
        at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:68)
        at dotty.tools.dotc.Run.compileUnits(Run.scala:253)
        at dotty.tools.dotc.Run.compileSources(Run.scala:186)
        at dotty.tools.dotc.Run.compile(Run.scala:170)
        at dotty.tools.dotc.Driver.doCompile(Driver.scala:35)
        at dotty.tools.dotc.Driver.process(Driver.scala:195)
        at dotty.tools.dotc.Driver.process(Driver.scala:163)
        at dotty.tools.dotc.Driver.process(Driver.scala:175)
        at dotty.tools.dotc.Driver.main(Driver.scala:205)
        at dotty.tools.dotc.Main.main(Main.scala)
```
</details>

--- a/compiler/src/dotty/tools/dotc/typer/Namer.scala
+++ b/compiler/src/dotty/tools/dotc/typer/Namer.scala
@@ -1072,33 +1072,17 @@ class Namer { typer: Typer =>
     /** The forwarders defined by export `exp` */
     private def exportForwarders(exp: Export)(using Context): List[tpd.MemberDef] =
       val buf = new mutable.ListBuffer[tpd.MemberDef]
-      val Export(expr, selectors0) = exp
+      val Export(expr, selectors) = exp
       if expr.isEmpty then
         report.error(em"Export selector must have prefix and `.`", exp.srcPos)
         return Nil
-
-      val renamed = mutable.Set[Name]()
-      val selectors = selectors0 map {
-        case sel @ ImportSelector(imported, id @ Ident(alias), bound) if alias != nme.WILDCARD =>
-          def noAliasSelector =
-            cpy.ImportSelector(sel)(imported, EmptyTree, bound).asInstanceOf[ImportSelector]
-          if renamed.contains(alias) then
-            report.error(i"duplicate rename target", id.srcPos)
-            noAliasSelector
-          else if alias == imported.name then
-            report.warning(i"redundant rename in export", id.srcPos)
-            noAliasSelector
-          else
-            renamed += alias
-            sel
-        case sel => sel
-      }
 
       val path = typedAheadExpr(expr, AnySelectionProto)
       checkLegalExportPath(path, selectors)
       lazy val wildcardBound = importBound(selectors, isGiven = false)
       lazy val givenBound = importBound(selectors, isGiven = true)
 
+      val targets = mutable.Set[Name]()
       def canForward(mbr: SingleDenotation, alias: TermName): CanForward = {
         import CanForward.*
         val sym = mbr.symbol
@@ -1108,8 +1092,8 @@ class Namer { typer: Typer =>
           Skip
         else if cls.derivesFrom(sym.owner) && (sym.owner == cls || !sym.is(Deferred)) then
           No(i"is already a member of $cls")
-        else if alias == mbr.name.toTermName && renamed.contains(alias) then
-          No(i"clashes with a renamed export")
+        else if targets.contains(alias) then
+          No(i"clashes with another export in the same export clause")
         else if sym.is(Override) then
           sym.allOverriddenSymbols.find(
             other => cls.derivesFrom(other.owner) && !other.is(Deferred)
@@ -1208,6 +1192,7 @@ class Namer { typer: Typer =>
         val size = buf.size
         val mbrs = List(name, name.toTypeName).flatMap(path.tpe.member(_).alternatives)
         mbrs.foreach(addForwarder(alias, _, span))
+        targets += alias
         if buf.size == size then
           val reason = mbrs.map(canForward(_, alias)).collect {
             case CanForward.No(whyNot) => i"\n$path.$name cannot be exported because it $whyNot"

--- a/tests/neg/i14818.check
+++ b/tests/neg/i14818.check
@@ -1,15 +1,16 @@
--- Error: tests/neg/i14818.scala:9:12 ----------------------------------------------------------------------------------
+-- Error: tests/neg/i14818.scala:9:17 ----------------------------------------------------------------------------------
 9 |  export M.{A, B as A}  // error
-  |            ^
-  |            no eligible member A at M
-  |            M.A cannot be exported because it clashes with a renamed export
+  |               ^^^^^^
+  |               no eligible member B at M
+  |               M.B cannot be exported because it clashes with another export in the same export clause
 -- [E050] Type Error: tests/neg/i14818.scala:16:10 ---------------------------------------------------------------------
 16 |  val x = b(1)  // error
    |          ^
    |          method b in object T3 does not take parameters
    |
    | longer explanation available when compiling with `-explain`
--- Error: tests/neg/i14818.scala:19:25 ---------------------------------------------------------------------------------
+-- Error: tests/neg/i14818.scala:19:22 ---------------------------------------------------------------------------------
 19 |  export M.{A as C, B as C} // error
-   |                         ^
-   |                         duplicate rename target
+   |                    ^^^^^^
+   |                    no eligible member B at M
+   |                    M.B cannot be exported because it clashes with another export in the same export clause

--- a/tests/neg/i14875.scala
+++ b/tests/neg/i14875.scala
@@ -1,0 +1,5 @@
+object M {
+  type B
+}
+
+export M.{B, B} // error

